### PR TITLE
Implement sorting for SelectMany in Frontend

### PR DESCRIPTION
**Pull request by brocgailit**
My PR is a:

💅 Enhancement 

Main update on the:

This PR adds sorting to SelectMany component in Content Manager utilizing react-sortable-hoc.

### CODE REVIEW

**Comment by lauriejim** [comment]
@brocgailit Can you join a screenshot ?

**Comment by soupette** [comment]
Hi @brocgailit thanks for contributing!
However, your PR introduces a bug with the `SelectMany`... I'm sure it won't be hard to debug!

Here are the step to reproduce (I cloned your repo):
- `npm run setup:build` at the root of the monorepo
- strapi new my-app --dev and go to `localhost:1337/admin`
- (I use mongo)
- create two content-types
  - product :
     - name
  - tag :
     - name
     - relation `many-to-many` with products
- add some tag entries
- create a product and try to associate a tag to it


Here's the error I got:

capture d ecran 2018-06-19 a 10 22 45

**Comment by brocgailit** [comment]
selectmany

(Apologies for the poor quality GIF)

**Comment by brocgailit** [comment]
@soupette is there is anything blocking the PR approval at this point?

**Comment by soupette** [comment]
Hi @brocgailit I tried your PR and it's now working.
However, we might need @Aurelsicoko approval as well on the UI.

Nice job

**Comment by Aurelsicoko** [comment]
I would prefer something like this if it's possible?

screen shot 2018-07-09 at 14 45 15

 If I take too much time I will improve the design by myself during a next release.

**Comment by brocgailit** [comment]
@Aurelsicoko I understand you're perspective for purely aesthetic reasons, however from a usability standpoint it is quite a nightmare.

This approach with a few linked entries may seem ideal, but when you include dozens (or more?) entries inline it causes significant reflow making it quite difficult to determine sorting order efficiently.  

Your approach also severely limits the extensibility of the Select Many component.  Consider a similar implementation to something like Contentful where you can easily interpret list order as well as extended action options like linking to edit the entry (Pencil icon).

image

**Comment by Aurelsicoko** [comment]
@brocgailit You're right! Let's go for this design, I will change the UI a little bit for follow our guidelines but the UX is better when there is a lot of entries to re-order 👍

**Comment by Aurelsicoko** [comment]
I've updated many things:

- New design
- New icons (sortable icon & remove icon)
- Remove SortableHandle HOC (the full width of the list's item can be dragged)
- Click on an item value allow you to jump directly to the Edit view of the item

File: packages/strapi-plugin-users-permissions/config/policies/permissions.js [hunk 1]
```diff
@@ -1,6 +1,8 @@
 module.exports = async (ctx, next) => {
   let role;
 
+  console.log("ljqskljdkls");
```
Comment (lauriejim) [inline packages/strapi-plugin-users-permissions/config/policies/permissions.js]: Can you remove this log please.

File: packages/strapi-plugin-users-permissions/models/User.settings.json [hunk 2]
```diff
@@ -38,6 +38,10 @@
       "via": "users",
       "plugin": "users-permissions",
       "configurable": false
+    },
```
Comment (lauriejim) [inline packages/strapi-plugin-users-permissions/models/User.settings.json]: Can you remove this relation.

**Comment by brocgailit** [comment]
image

Looks like overflow is being set to scroll on the ul element which is causing overlap here.  I like this approach otherwise.  Good work!

**Pull request merged**